### ISSUE TITLE
Add dependencies

### ISSUE THREAD

**Issue by marissahuysentruyt** [marissahuysentruyt/estimate-tracker#2]
As a developer, linting, testing and TypeScript will be very handy while building the app, so add those dependencies to get the project set up correctly from the start

## Acceptance Criteria
- add linters (eslint, stylelint)
- add testing dependencies (cypress & vitest)
- set up with vite for compiling (? if necessary with chosen framework)
- set up with typescript/scss
- add linting, testing, and dev/start scripts to `package.json`

**Comment by marissahuysentruyt** [comment]
estimated time: 1hr

**Comment by marissahuysentruyt** [comment]
actual time: 3.5 hrs

_Lots of these were part of the project setup with `npm create vue@latest`. The configs for TypeScript, eslint, vitest, cypress were all pre-configured, and Vue compiles everything with Vite by default. I misunderstood that Vite was able to automatically compile SCSS, without additional configuration. So I went down a rabbit hole of trying to create `build:scss` and `watch:scss` commands, to try to add to the `build` script for compiling all of it. All I needed to do, unfortunately was `@import` the main stylesheet into the `main.ts` file. Quite a bit of time was wasted _
